index.js

```diff
@@ -1,19 +1,85 @@
-var gumshoe = require('gumshoe')
+const Promise = require('bluebird');
+const debug = require('debug');
+const glob = Promise.promisify(require('glob'));
+const readFile = Promise.promisify(require('fs').readFile);
+const path = require('path');
 
-// *rules* - list of detection rules
-// *phase* - which phase to run matches for (prepare, test, deploy, cleanup)
-// *ctx* - Strider worker plugin context
-// *cb* - callback function, takes
-module.exports = function(rules, phase, ctx, cb) {
+/**
+ * Resolve property on context.
+ */
+function resolve(context, property) {
+  return property.split('.')
+    .reduce((result, key) => (result ? result[key] : undefined), context);
+}
+
+/**
+ * Parse a file and return rule if it matches.
+ * @param {string} file file name
+ * @param {object} rule the rule to match
+ */
+function parseFile(file, rule) {
+  return readFile(file, 'utf8')
+    .then((data) => {
+      let result;
+      if (rule.grep && rule.grep.exec(data) !== null) {
+        result = true;
+      }
+      if (result !== false && rule.jsonKeyExists) {
+        try {
+          result = resolve(JSON.parse(data), rule.jsonKeyExists) !== undefined;
+        } catch (e) {
+          result = false;
+        }
+      }
+      return (result === true) ? rule : undefined;
+    })
+    .catch(() => undefined);
+}
 
-  // Note: we only run the first matched result.
-  // There could be more, but we ignore them at the moment.
-  gumshoe.run(ctx.workingDir, rules, function(err, result) {
-      if (err) return cb(0)
-      var psh = ctx.shellWrap(result[phase])
-      ctx.forkProc(ctx.workingDir, psh.cmd, psh.args, function(code) {
-        return cb(code)
-      })
-  })
+/**
+ * Detect and run rule from a set of rules.
+ * @param {object} rules list of detection rules
```

Comment:
This should be array

Reply:
Yes. Sorry for the late response. Keeping it now since already merged.